### PR TITLE
Add ability to delete a token

### LINKED ISSUE #4234: missing route DELETE /users/{username}/tokens
- Gitea version (or commit ref): 1.4.2
- Git version: 2.7.4
- Operating system: `CentOS release 6.9 (Final)`
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

There are two API routes for the user auth token in the current swagger docs: https://try.gitea.io/api/swagger

```
GET /users/{username}/tokens 
    List the authenticated user's access tokens

POST /users/{username}/tokens 
    Create an access token
```

but `DELETE` a token for a user is missing.  
See https://github.com/go-gitea/gitea/blob/908e8942ccae5b7966c7084780b3441e2190d9c1/routers/api/v1/user/app.go

So, what it needs (next to the swagger definitions) is, that the  function  
`func DeleteAccessToken(ctx *context.APIContext)`   
in  
`gitea/routers/api/v1/user/app.go`   
is added, which calls `models.DeleteAccessTokenByID(token_id, user_id)`, right?

Or is this route for some reason an unwanted feature?

### PR DESCRIPTION
**Pull request by techknowlogick**
Fix #4234

~Edit: Blocked by https://github.com/go-gitea/go-sdk/pull/110~

### CODE REVIEW

File: routers/api/v1/user/app.go
Comment (techknowlogick) [inline routers/api/v1/user/app.go:102]: Question for reviewers: This deletes based on ID (int), however we only return the name (string) of the token above. So this works, but users need to know what the ID (int) is before using this. My question is: Should this be changed to delete based on token name (string)?

Comment (lafriks) [reply routers/api/v1/user/app.go:102]: How does GitHub API works?

Comment (techknowlogick) [reply routers/api/v1/user/app.go:102]: @lafriks Their docs say they use the integer ID of the token: https://developer.github.com/v3/oauth_authorizations/#delete-an-authorization

They also list the ID in the response when listing the tokens, so I will keep this PR the same, and I'll update this PR to include the ID in the List endpoint.

**Review by sapk — APPROVED** [review]
Need to be rebase after go-gitea/go-sdk#110 get merged but LGTM

**Review by sapk — APPROVED** [review]
Need to be rebase after go-gitea/go-sdk#110 get merged but LGTM

**Comment by techknowlogick** [comment]
As dependency has now been merged it is ready for testing & review

**Review by sapk — APPROVED** [review]

**Comment by lunny** [comment]
The API seems not compatible.

**Comment by techknowlogick** [comment]
@lunny I'm not sure what you mean. I was able to see that ID is now a part of the API response
```
[{"id":3,"name":"drone","sha1":"1bf6fa0e0c685ef2cb4448c72891ab0efe888343"},{"id":2,"name":"test","sha1":"337466149517e73ecc47df9aa18696dc897d5518"}]
```
and that I can also send DELETE requests to the tokens endpoint (using basic-auth to match GitHub behaviour)

**Comment by jonasfranz** [comment]
Integration tests would be good.

**Comment by techknowlogick** [comment]
@JonasFranzDEV tests added, and they pass.

**Review by lafriks — APPROVED** [review]

**Pull request merged**
